src/core/server/config/schema/types/__tests__/string_type.test.ts

```diff
@@ -53,6 +53,42 @@ describe('#maxLength', () => {
   });
 });
 
+describe('#hostname', () => {
+  test('returns value for valid hostname as per RFC1123', () => {
+    const hostNameSchema = schema.string({ hostname: true });
+
+    expect(hostNameSchema.validate('www.example.com')).toBe('www.example.com');
```

Comment:
__note:__ test cases taken from [Joi tests](https://github.com/hapijs/joi/blob/master/test/types/string.js#L959) + 1 for Punycode